### PR TITLE
Migrate assets to dll

### PR DESCRIPTION
**Pull request by Shawdooow**

### CODE REVIEW

File: osu-framework/osu.Framework/Allocation/DependencyContainer.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ private void register(Type type, bool lazy)
             {
                 var permitNull = initializer.GetCustomAttribute<BackgroundDependencyLoader>().PermitNulls;
                 var parameters = initializer.GetParameters().Select(p => p.ParameterType)
-                                            .Select(t => (Func<object>)(() =>
+                                            .Select(t => new Func<object>(() =>
```
Comment (Jorolf) [inline osu-framework/osu.Framework/Allocation/DependencyContainer.cs]: why do you have to cast it anyway?

Comment (Jorolf) [inline osu-framework/osu.Framework/Allocation/DependencyContainer.cs]: oh wait
thats the framework

File: osu.Game.Rulesets.Vitaru/osu.Game.Rulesets.Vitaru.csproj [hunk 2]
```diff
@@ -105,7 +105,36 @@
...
+  <ItemGroup>
+    <EmbeddedResource Include="Assets\Textures\playerKiai.png" />
+  </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="Assets\Textures\sign.png" />
+  </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="Assets\Audio\Samples\deathSound.mp3" />
+  </ItemGroup>
+  <ItemGroup>
```
Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/osu.Game.Rulesets.Vitaru.csproj]: what heppened here?

Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/osu.Game.Rulesets.Vitaru.csproj]: it includes deathSound as an mp3?

Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/osu.Game.Rulesets.Vitaru.csproj]: no I meant why there's an ItemGroup for every Resource

File: osu-framework/osu.Framework/Allocation/DependencyContainer.cs (hunk 1 above)
Comment (peppy) [inline osu-framework/osu.Framework/Allocation/DependencyContainer.cs]: want to PR this stuff to the framework?

File: osu-framework/osu.Framework/Extensions/ExtensionMethods.cs [hunk 3]
```diff
@@ -37,7 +37,7 @@ public static T Find<T>(this List<T> list, Predicate<T> match, int startIndex)
 
             int val = list.FindIndex(startIndex, list.Count - startIndex - 1, match);
 
-            return list.ValueAtOrDefault(val);
+            return list.ElementAtOrDefault(val);
```
Comment (peppy) [inline osu-framework/osu.Framework/Extensions/ExtensionMethods.cs]: this too

File: osu-framework/osu.Framework/Extensions/ExtensionMethods.cs [hunk 4]
```diff
@@ -87,18 +87,6 @@ public static bool IsValidIndex<T>(this List<T> list, int index)
         }
 
         /// <summary>
-        /// Validates whether index is valid, before returning the value at the given index.
-        /// </summary>
-        /// <typeparam name="T">Probably should limit to nullable types.</typeparam>
-        /// <param name="list">The list to take values</param>
-        /// <param name="index">The index to request values from</param>
-        /// <returns>Value at index, else the default value</returns>
-        public static T ValueAtOrDefault<T>(this List<T> list, int index)
```
Comment (peppy) [inline osu-framework/osu.Framework/Extensions/ExtensionMethods.cs]: this too

File: osu-framework/osu.Framework/Allocation/DependencyContainer.cs (hunk 1 above)
Comment (Shawdooow) [inline osu-framework/osu.Framework/Allocation/DependencyContainer.cs]: Is it not part of the framework? I thought this came from updating the fork to latest osu?

Comment (peppy) [inline osu-framework/osu.Framework/Allocation/DependencyContainer.cs]: okay you need to start using submodules and doing things the right way then.

**Pull request merged**